tests/shared/nlu/training_data/test_training_data.py

```diff
@@ -295,6 +297,87 @@ def test_train_test_split(filepaths: List[Text]):
         )
 
 
+def test_number_of_examples_per_intent():
+    message_action = Message(data={"action_name": "utter_greet"})
+    message_intent = Message(
+        data={"text": "I would like the newsletter", "intent": "subscribe"}
+    )
+    message_non_nlu_intent = Message(data={"intent": "subscribe"})
+    message_other_intent_one = Message(
+        data={"text": "What is the weather like today?", "intent": "ask_weather"}
+    )
+    message_other_intent_two = Message(
+        data={"text": "Will it rain today?", "intent": "ask_weather"}
+    )
+    message_non_nlu_other_intent_three = Message(data={"intent": "ask_weather"})
+
+    training_examples = [
+        message_action,
+        message_intent,
+        message_non_nlu_intent,
+        message_other_intent_one,
+        message_other_intent_two,
+        message_non_nlu_other_intent_three,
+    ]
+    training_data = TrainingData(training_examples=training_examples)
+
+    assert training_data.number_of_examples_per_intent["subscribe"] == 1
+    assert training_data.number_of_examples_per_intent["ask_weather"] == 2
+
+
+def test_number_of_examples_per_intent_with_yaml():
+    stories = rasa.shared.utils.io.read_yaml_file(
+        "data/test_number_nlu_examples/stories.yml"
+    )
+    rules = rasa.shared.utils.io.read_yaml_file(
+        "data/test_number_nlu_examples/rules.yml"
+    )
+    stories.update(rules)
+
+    non_nlu_data = []
+    for v in stories.values():
+        for i in range(len(v)):
+            for d in v[i]["steps"]:
+                if d.get("intent") is not None:
+                    non_nlu_data.append(Message(data=d))
+
+    non_nlu_td = TrainingData(training_examples=non_nlu_data)
+    assert non_nlu_td.number_of_examples_per_intent.get("greet") is None
+    assert non_nlu_td.intents == {"greet", "ask_weather"}
+
+    from rasa.shared.importers.utils import training_data_from_paths
+
+    td = training_data_from_paths(
+        ["data/test_number_nlu_examples/nlu.yml"], language="en"
+    )
+    td.merge(non_nlu_td)
```

Comment:
A easier and more realistic way to get a combined training data from the nlu, stories and rules could instead be something like:
```
    importer = TrainingDataImporter.load_from_dict(
        {}, str(config_path), str(domain_path), ["data/test_number_nlu_examples/nlu.yml", "data/test_number_nlu_examples/stories.yml", "data/test_number_nlu_examples/rules.yml"]
    )
    training_data = await importer.get_nlu_data()
```

(See `test_without_additional_e2e_examples` for how to create empty domain config files.)

Reply:
Neat! Actioned 😃